lib/lsperfm/core/stats.rb

```diff
@@ -0,0 +1,64 @@
+# encoding: utf-8
+require "thread"
+
+Thread.abort_on_exception = true
+
+module LogStash::PerfM
+
+  class Stats
+
+    REFRESH_COUNT = 100
+
+    attr_accessor :stats
+
+    def initialize
+      @stats = []
+    end
+    # below stats counter and output reader threads are sharing state using
+    # the @stats_lock mutex, @stats_count and @stats. this is a bit messy and should be
+    # refactored into a proper class eventually
+
+    def detach_stats_counter
+      Thread.new do
+        loop do
+          start = @stats_lock.synchronize{@stats_count}
+          sleep(1)
+          @stats_lock.synchronize{@stats << (@stats_count - start)}
+        end
+      end
+    end
+
+    # detach_output_reader spawns a thread that will fill in the @stats instance var with tps samples for every seconds
+    # @stats access is synchronized using the @stats_lock mutex but can be safely used
+    # once the output reader thread is completed.
+    def detach_output_reader(io, regex)
+      Thread.new(io, regex) do |io, regex|
+        i = 0
+        @stats = []
+        @stats_count = 0
+        @stats_lock = Mutex.new
```

Comment:
This method is intended to be called only once while running a test, so I guess the code is good.

On the other side, I don't dislike the idea of having the mutex as ```@stats_lock```  it makes the thing more expressive. 

I guess we can improve this class in the long run, but for now I don't particularly see problems in here.

Do you see more potential problems in this class?